plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/SimpleBuiltinCostModel.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+{- | A program to parse a JSON representation of costing functions for Plutus Core
+   builtins and and produce a simple cost model which can be used from Agda and other
+   executables -}
+module PlutusCore.Evaluation.Machine.SimpleBuiltinCostModel
+   ( BuiltinCostMap
+   , defaultSimpleBuiltinCostModel
+   ) where
+
+
+import Data.Aeson.Key as Key (toText)
+import Data.Aeson.KeyMap qualified as KeyMap
+import Data.Aeson.THReader (readJSONFromFile)
+import Data.Bifunctor
+import Data.Text (Text, replace)
+import PlutusCore.DataFilePaths qualified as DFP
+import PlutusCore.Evaluation.Machine.CostingFun.SimpleJSON
+
+type BuiltinCostMap = [(Text, CpuAndMemoryModel)]
+
+-- | The default builtin cost map.
+builtinCostKeyMap :: KeyMap.KeyMap CpuAndMemoryModel
```

Comment:
I wonder if it would be cleaner to instead create this from the read-in Haskell version of the cost model that we already have, rather than having two independent JSON readers?

Reply:
That would also mean we could make it from a `CostModelParams` by going through the full version.

Reply:
If you are suggesting how to convert from the Haskell model to the model I need, I wouldn't know how to do it other than the naive way of going through every possible builtin explicitly.

Reply:
Hmmm. I guess that is basically what you'd have to do, or do some Generics magic. Annoying. But it feels bad duplicating the JSON format...

Reply:
The JSON format was already duplicated in the print-cost-model executable. So if we unify the JSON parser in that executable and the one used here, we have no more duplication than before.